src/mcfret.c

```diff
@@ -0,0 +1,649 @@
+      exit(0);
+    }
+    rate_matrix=(float *)calloc(segments*segments,sizeof(float));
+
+  /* Tell the user that we are in the MCFRET Routine */
+     if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Autodetect")) || (!strcmp(non->technique, "MCFRET-Absorption"))
+      || (!strcmp(non->technique, "MCFRET-Emission")) || (!strcmp(non->technique, "MCFRET-Coupling")) || (!strcmp(non->technique,
+      "MCFRET-Rate")) || (!strcmp(non->technique, "MCFRET-Analyse")) ) {
+          printf("Perfroming the MCFRET calculation.\n");
+    }
+
+/* Call the absorption routine */
+    if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Absorption"))){
+        mcfret_response_function(re_Abs,im_Abs,non,0);
+    }
+
+/* Call the emission routine */
+    if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Emission"))){
+        mcfret_response_function(re_Emi,im_Emi,non,1);
+    }
+
+/* Call the coupling routine */
+    if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Coupling"))){
+        mcfret_coupling(J,non);
+    }
+
+/* Call the rate routine routine */
+    if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Rate"))){
+        if ((!strcmp(non->technique, "MCFRET-Rate"))){
+            /* Read in absorption, emission and coupling from file if needed */
+        }
+        mcfret_rate(rate_matrix,segments,re_Abs,im_Abs,re_Emi,im_Emi,J,non);
+    }
+
+/* NOTE!!! We still need to write the calculated stuff to file */
+    write_matrix_to_file("RateMatrix.dat",rate_matrix,segments);
+
+    free(re_Abs);
+    free(im_Abs);
+    free(re_Emi);
+    free(im_Emi);
+    free(J);
+    free(rate_matrix);
+    return;
+}
+
+/* Calculate Absorption/Emission matrix (depending on emission variable 0/1) */
+void mcfret_response_function(float *re_S_1,float *im_S_1,t_non *non,int emission){
+/* Define variables and arrays */
+   /* Integers */
+    int nn2;
+    int itime,N_samples;
+    int samples;
+    int x,ti,tj,i,j;
+    int t1;
+    int elements;
+    int cl,Ncl;
+
+    /*Hamiltonian of the whole system - all donors and acceptors included*/
+    float *Hamil_i_e;
+    /*Vectors representing time dependent states: real and imaginary part*/
+    float *vecr, *veci;
+    float shift1; 
+
+  /* Time parameters */
+    time_t time_now,time_old,time_0;
+  /* Initialize time */
+    time(&time_now);
+    time(&time_0);
+    shift1=(non->max1+non->min1)/2;
+    printf("Frequency shift %f.\n",shift1);
+    non->shifte=shift1;
+
+  /*File handles*/
+    FILE *H_traj;
+    FILE *C_traj;
+    /* FILE *absorption_matrix, *emission_matrix,*/
+    FILE *log;
+    FILE *Cfile;
+
+    /*Allocate memory for all the variables */
+    /*Allocating memory for the real and imaginary part of the wave function that we need to propagate*/
+    vecr=(float *)calloc(non->singles*non->singles,sizeof(float));	
```

Comment:
You do define a variable `nn2` but don't use it here for the size of `vecr` and `veci`.